tests/test_base.py

```diff
@@ -69,3 +71,18 @@ def mock_getsyspath(path):
 
         with self.assertRaises(errors.InvalidPath):
             self.fs.validatepath("0123456789A")
+
+    def test_quote(self):
+        test_fixtures = [
+            # test_snippet, expected
+            ["foo/bar/egg/foofoo", "foo/bar/egg/foofoo"],
+            ["foo/bar ha/barz", "foo/bar%20ha/barz"],
+            ["example b.txt", "example%20b.txt"],
+            ["exampleㄓ.txt", "example%E3%84%93.txt"],
+        ]
+        if platform.system() == "Windows":
```

Comment:
As mentioned above, might be worth also testing paths with `:` and `\` on non-Windows systems?